include/powsybl/iidm/Network.hpp

```diff
@@ -161,6 +163,10 @@ class Network : public Container, public VariantManagerHolder {
     template <typename T = Connectable, typename = typename std::enable_if<std::is_base_of<Connectable, T>::value>::type>
     stdcxx::range<T> getConnectables();
 
+    const ConnectedComponentsManager& getConnectedComponentsManager() const;
+
+    ConnectedComponentsManager& getConnectedComponentsManager();
+
```

Comment:
These methods should be private as they are not in Java api

Reply:
It's true, but this method is needed by a lot of classes in the implementation, so I have to declare lot of classes as friend. I'm not sure it's better. The ComponentManager provides only one risky method which is the `invalidate()`. This will lead to a complete calculation of the connectivity.

It seems acceptable to keep these methods public.